cmd/agent/subcommands/analyzelogs/command.go

```diff
@@ -108,27 +133,56 @@ func runAnalyzeLogs(cliParams *CliParams, config config.Component) error {
 }
 
 // Used to make testing easier
-func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider) {
+func runAnalyzeLogsHelper(cliParams *CliParams, config config.Component, ac autodiscovery.Component) (chan *message.Message, *launchers.Launchers, pipeline.Provider) {
 	configSource := sources.NewConfigSources()
-	wd, err := os.Getwd()
+	waitTime := time.Duration(1) * time.Second
+	waitCtx, cancelTimeout := context.WithTimeout(
+		context.Background(), waitTime)
+	common.LoadComponents(nil, nil, ac, pkgconfigsetup.Datadog().GetString("confd_path"))
+	ac.LoadAndRun(context.Background())
+	allConfigs, err := common.WaitForConfigsFromAD(waitCtx, []string{cliParams.LogConfigPath}, 1, "", ac)
+	cancelTimeout()
 	if err != nil {
-		fmt.Println("Cannot get working directory")
 		return nil, nil, nil
 	}
-	absolutePath := wd + "/" + cliParams.LogConfigPath
-	data, err := os.ReadFile(absolutePath)
-	if err != nil {
-		fmt.Println("Cannot read file path of logs config")
-		return nil, nil, nil
+	var sources []*sources.LogSource
```

Comment:
runAnalyzeLogsHelper is doing a lot of stuff now. Perhapse we can break out the config resolution logic to it's own helper function? ex something like `func getSources() []*sources.LogSource` that encapsulates all the source resolution.